### PR TITLE
ruin nicks day; initial commit

### PR DESCRIPTION
**Pull request by c0nrad**

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	if err != nil {
+		return "anonymous"
+	}
+
+	return c.Value
+}
+
+func NewPolicyHandler(w http.ResponseWriter, r *http.Request) {
+	policy, err := NewPolicy("New Policy")
+
```
Comment (ncapalbo) [inline api.go]: nit: remove newline to keep formatting consistent with the methods above/below.

File: datastore.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	} else {
+		Session = ConnectToMongo("localhost")
+	}
+}
+
+func ConnectToMongoTLS(uri string) *mgo.Session {
+	tlsConfig := &tls.Config{}
+	tlsConfig.InsecureSkipVerify = true
+
+	dialInfo, err := mgo.ParseURL(uri)
```
Comment (ncapalbo) [inline datastore.go]: Please handle err here in some fashion.

File: datastore.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	}
+
+	session, err := mgo.DialWithInfo(dialInfo)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Connect to mongo")
+
+	// EnsureUserIndex(session)
```
Comment (ncapalbo) [inline datastore.go]: Here and below, please remove commented code.

File: datastore.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	DB = "security-metrics"
+
+	StartingBalance = 100
+)
+
+var MongoUri = ""
+var Session *mgo.Session
+
+func init() {
+	MongoUri = os.Getenv("MONGO_URI")
```
Comment (ncapalbo) [inline datastore.go]: If I'm following the flow correctly, you don't need to declare MongoUri as a global.  os.Getenv [1] returns nil if the variable isn't present.  

[1] https://golang.org/pkg/os/#Getenv

File: main.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/codegangsta/negroni"
+)
+
+var (
```
Comment (ncapalbo) [inline main.go]: Is there a reason this is var and not const?

File: datastore.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+	if MongoUri != "" {
```
Comment (ncapalbo) [inline datastore.go]: Initialisms should never be Uri, but either URI or uri [1]

[1] https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: datastore.go [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"os"
+
+	mgo "gopkg.in/mgo.v2"
+)
+
+const (
```
Comment (ncapalbo) [inline datastore.go]: Can you please add a comment as to what each of these are?  The collections are obvious but the last is not.  Please use full sentences [1].

[1] https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

File: models.go [hunk 8]
```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"errors"
+	"time"
+
+	"gopkg.in/mgo.v2/bson"
+)
+
+type Grade struct {
```
Comment (ncapalbo) [inline models.go]: Small style suggestion: you can clean up the struct by grouping struct fields by their type.  For example:
PolicyID, StatementID, ProjectID bson.ObjectId

File: models.go [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+		return nil, errors.New("not a valid objectid for policy")
+	}
+
+	grades := make([]*Grade, 0)
+
+	err := session.DB(DB).C(GradeCollection).Find(bson.M{"projectid": bson.ObjectIdHex(project), "policyid": bson.ObjectIdHex(policy)}).Sort("ts").All(&grades)
+	return grades, err
+}
+
+type Project struct {
```
Comment (ncapalbo) [inline models.go]: Same thing here with fields of the same type

File: api.go (hunk 1 above)
Comment (c0nrad) [reply api.go]: fixed

File: datastore.go (hunk 7 above)
Comment (c0nrad) [reply datastore.go]: done

File: datastore.go (hunk 4 above)
Comment (c0nrad) [reply datastore.go]: ah nice catch. legacy carry over.

File: datastore.go (hunk 6 above)
Comment (c0nrad) [reply datastore.go]: fixed

File: datastore.go (hunk 2 above)
Comment (c0nrad) [reply datastore.go]: whoops, nice catch

File: datastore.go (hunk 3 above)
Comment (c0nrad) [reply datastore.go]: removed

File: main.go (hunk 5 above)
Comment (c0nrad) [reply main.go]: d-d-d-dynamic web ports. You ganna find the port if you want to use it.

(fixed)

File: models.go (hunk 8 above)
Comment (c0nrad) [reply models.go]: it's in that grouping for a reason. added comments to explain why

File: models.go (hunk 9 above)
Comment (c0nrad) [reply models.go]: fixed!

File: datastore.go [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+func ConnectToMongo(uri string) *mgo.Session {
+	session, err := mgo.Dial(uri)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Connect to mongo")
+	// EnsureUserIndex(session)
```
Comment (ncapalbo) [inline datastore.go]: nit: please remove

Comment (c0nrad) [reply datastore.go]: removed m'lord.

**Pull request merged**